src/json.hpp

```diff
@@ -8800,64 +8800,187 @@ class basic_json
             return result;
         }
 
+        // non locale aware isspace
+        bool nl_isspace(const char c) const
+        {
+            return
+                c == 0x20 or
+                c == 0x09 or
+                c == 0x0a or
+                c == 0x0b or
+                c == 0x0c or
+                c == 0x0d;
+        }
+
+        // non locale aware isdigit
+        // Microsoft in 1252 codepage and others may classify additional single-byte characters as digits using std::isdigit
+        bool nl_isdigit(const char c) const
+        {
+            return c >= '0' and c <= '9';
+        }
+
         /*!
-        @brief parse floating point number
+        @brief parse string to floating point number
+
+        This function is a reimplementation of the strtold family without
+        regard to locale
 
-        This function (and its overloads) serves to select the most approprate
-        standard floating point number parsing function based on the type
-        supplied via the first parameter.  Set this to @a
-        static_cast<number_float_t*>(nullptr).
+        @tparam T  a is_floating_point type
 
-        @param[in] type  the @ref number_float_t in use
+        @param[in] st  the string we will parse
 
         @param[in,out] endptr recieves a pointer to the first character after
         the number
 
         @return the floating point number
         */
-        long double str_to_float_t(long double* /* type */, char** endptr) const
+        template <typename T, typename = typename std::enable_if<
+            std::is_floating_point<T>::value>::type>
+        T strtox(const char *st, char **endptr) const
         {
-            return std::strtold(reinterpret_cast<typename string_t::const_pointer>(m_start), endptr);
-        }
+            constexpr std::array<long double, 9> powerof10 {
+                {1.e1L, 1.e2L, 1.e4L, 1.e8L, 1.e16L, 1.e32L, 1.e64L, 1.e128L, 1.e256L}
+            };
 
-        /*!
-        @brief parse floating point number
+            T result = 0;
+            const char *fst = st;
+            bool successful_parse = false;
 
-        This function (and its overloads) serves to select the most approprate
-        standard floating point number parsing function based on the type
-        supplied via the first parameter.  Set this to @a
-        static_cast<number_float_t*>(nullptr).
+            while (nl_isspace(*fst))
```

Comment:
Whitespace is already skipped by the lexer, so I think it is safe to skip this loop.